### ISSUE TITLE
Find-in-Files "Filters" box of ONLY "exclusion" items finds no matches; should auto-add *.*

### ISSUE THREAD

**Issue by sasumner** [notepad-plus-plus/notepad-plus-plus#8169]
### Description of the Issue
Notepad++ 7.8.2 added the ability to use "exclusion" specs in the _Filters_ box for _Find in Files_.

Example: Put `!*.log` in _Filters_ to exclude files ending with `.log` from the search.  Multiple exclusion specs can be used, such as `!*.log !*.obj` to suppress searching of log and object files.

However, as pointed out here:

https://community.notepad-plus-plus.org/topic/18609/question-regarding-new-find-in-files-filter-excluding-ability-in-version-7-8-2

using **only** "exclusion" specs results in no possible matches.

The technique that works is to put `*.* !*.log` in the _Filters_ box.  Note the addition of the `*.*`

But, one can leave the _Filters_ box totally empty and the effective filter spec will be `*.*`, so why shouldn't the case of "only exclusion filters" also infer a `*.*`?

### Steps to Reproduce the Issue
1. Perform any _Find in Files_ search leaving the _Filters_ box empty.  Note results; the assumption is that you search for something that will generate some number of hits.
1a. Note also that after the search, Notepad++ has put `*.*` in the _Filters_ box for you.  How nice.
2. Remove the `*.*` and put _only_ `!*.zzz` in the _Filters_ box, where `zzz` is any file extension that you don't have present in the directory tree of the search.
3. Run the _Find in Files_ search again.

### Expected Behavior

I expected the same hits from the search in step 3 as were obtained in step 1.

### Actual Behavior

No hits occur, i.e. `(0 hits in 0 files)` .

**BOTTOM LINE FOR THIS ISSUE**:  Using only "exclusion" specs in the _Filters_ box should result in a search of all files except the exclusion matches.

### Debug Information
Notepad++ v7.8.6   (32-bit)
Build time : Apr 21 2020 - 15:17:06
Path : C:\.............\npp.7.8.6.bin\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 1809
OS Build : 17763.1039
Plugins : mimeTools.dll NppConverter.dll NppExport.dll

**Comment by donho** [comment]
@sasumner Could you provide a complete description (or a small specification) for the (your) expected behaviour?

**Comment by sasumner** [comment]
@donho Your example is good.  I don't think the program needs to add `*.*` to the box when the box is not empty.  It _could_, if you like it that way, but not required. 

My creating this issue was based on user feedback on Community that I found reasonable:

 _**"Note that you cannot just write an exclusive mask, alone ! For instance, the `!*.log` filter does not scan any file :-(( Personally, I think that `!*.log` should, implicitly, mean `!*.log *.*` i. e. it would have scanned all files, except for those with extension = log"**_

**Comment by sasumner** [comment]
@donho If you want to assign it over to me, I can implement the issue.  Or, if you are already working on it, that's fine.

**Comment by donho** [comment]
> I don't think the program needs to add *.* to the box when the box is not empty. It could, if you like it that way, but not required.

So is it the behaviour you expect ?
```
<EMPTY> -> find in all, and full fill *.* on the UI for the users
*.* -> find in all
*.* !*.log !*.exe-> find in all except log & exe files
!*.exe  -> find in all except exe files
```

**Comment by sasumner** [comment]
@donho 

Yes.
